File: Data Structures & Algorithms/jump-game-ii/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    jump(nums) {
        let minJump = Number.MAX_SAFE_INTEGER
        function jumps(index,count){
            if(index === nums.length-1){
                minJump = Math.min(minJump,count)
                return
            }
            let farthest = Math.min(nums[index]+index,nums.length-1)
            for(let i = farthest;i>index;i--){
                    jumps(i,count+1)
            }
            return
        }
        jumps(0,0)
        return minJump
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.java
class Solution {
    public int findDuplicate(int[] nums) {
        int[] map=new int[nums.length];
        for(int i :nums){
            map[i-1]+=1;
        }
        for (int i = 0; i < map.length; i++) {
            if(map[i]>=2){
                return i+1;
            }
        }
        return 0;
    }
}
